pkg/aws/core.go

```diff
@@ -160,12 +160,15 @@ func (d *Driver) CreateMachine(ctx context.Context, req *driver.CreateMachineReq
 		MinCount:            aws.Int64(1),
 		MaxCount:            aws.Int64(1),
 		UserData:            &UserDataEnc,
-		KeyName:             aws.String(providerSpec.KeyName),
 		IamInstanceProfile:  iam,
 		NetworkInterfaces:   networkInterfaceSpecs,
 		TagSpecifications:   []*ec2.TagSpecification{tagInstance, tagVolume, tagNetworkInterface},
 	}
 
+	if len(providerSpec.KeyName) > 0 {
```

Comment:
Since `KeyName` is not optional, shouldn't the `AwsProviderSpec` be update to reflect that?
```go
type AWSProviderSpec struct {
   ...
   // +optional
   KeyName *string
}
```
Will this cause an issue for any consumer?

Reply:
The `// +optional` is not needed since it is not used for the other optional fields in the struct `CapacityReservationTarget` and `SpotPrice`. Changing `KeyName` to be a pointer would be reasonable, however that would be a breaking change for whoever has imported this struct and I am not familiar with the consumers.

Reply:
We checked that `AwsProviderSpec` is not used outside this repo. So changing it to pointer to string should be fine.
@himanshu-kun would you also agree?

Reply:
Yes I agree, we should do that. Please adapt the release note to note the field name change as `breaking change`